spec/index.bs

```diff
@@ -4241,19 +4238,32 @@ dictionary GPUFragmentState: GPUProgrammableStage {
 
 <div algorithm>
     <dfn abstract-op>validating GPUFragmentState</dfn>(|descriptor|)
-        Return `true` if all of the following conditions are satisfied:
+        Return `true` if all of the following requirements are met:
+
+        - |descriptor|.{{GPUFragmentState/targets}}.length must be &le; 4.
+        - For each |colorState| layout descriptor in the list |descriptor|.{{GPUFragmentState/targets}}:
+            - |colorState|.{{GPUColorTargetState/format}} must be listed in [[#plain-color-formats]]
+                with {{GPUTextureUsage/RENDER_ATTACHMENT}} capability.
+            - If |colorState|.{{GPUColorTargetState/blend}} is not `undefined`:
+                - The |colorState|.{{GPUColorTargetState/format}} must be filterable
+                    according to the [[#plain-color-formats]] table.
+                - |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}
+                    must be a [=valid GPUBlendComponent=].
+                - |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/alpha}}
+                    must be a [=valid GPUBlendComponent=].
+            - |colorState|.{{GPUColorTargetState/writeMask}} must be &lt; 16.
+            - |descriptor|.{{GPUProgrammableStage/module}} must contain an output variable that:
+                - is [=statically used=] by |descriptor|.{{GPUProgrammableStage/entryPoint}}, and
+                - has a type that is compatible with |colorState|.{{GPUColorTargetState/format}}.
```

Comment:
Note for this PR, but we probably need better specification of what compatible means here.